File: output/gpt_4o/avatar/algo_based_translation/python/java/failed_translation/atcoder_ABC168_C.java

import java.util.Scanner;
import java.lang.Math;

public class atcoder_ABC168_C {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Parse input
        int a = scanner.nextInt();
        int b = scanner.nextInt();
        int h = scanner.nextInt();
        int m = scanner.nextInt();
        
        // Calculate the angle `C` in degrees between the hour and minute hand
        double angle_minute_hand = 360.0 / 60 * m;
        double angle_hour_hand = 360.0 / 12 * h + 360.0 / 12 / 60 * m;
        double C = Math.abs(angle_minute_hand - angle_hour_hand);
        
        // Normalize the angle `C` to be the smaller angle between the two hands
        if (C > 180) {
            C = 360 - C;
        }
        
        // Convert angle `C` to radians
        double angle_C_radians = Math.toRadians(C);
        
        // Calculate the cosine of angle `C`
        double cosC = Math.cos(angle_C_radians);
        
        // Apply the law of cosines to find the distance between the two ends of the clock hands
        double distance = Math.sqrt(a * a + b * b - 2 * a * b * cosC);
        
        // Output the calculated distance
        System.out.println(distance);
        
        scanner.close();
    }
}

File: output/gpt_4o/avatar/algo_based_translation/java/python/failed_translation/atcoder_ABC168_C.py

import math
from decimal import Decimal, getcontext

def main():
    getcontext().prec = 50  # Set precision for Decimal calculations

    # Input operations
    A = int(input())
    B = int(input())
    H = int(input())
    M = int(input())

    # Convert to Decimal
    AA = Decimal(A)
    BB = Decimal(B)
    HH = Decimal(H)
    MM = Decimal(M)

    # Calculations
    mkaku = MM * 6
    hkaku = HH * 30 + (Decimal(M) / 2)
    kaku = mkaku - hkaku
    dkaku = math.radians(abs(kaku))

    AA2 = AA * AA
    BB2 = BB * BB
    CC = AA2 + BB2
    DD = 2 * math.cos(dkaku) * AA * BB
    ans2 = CC - DD
    dans2 = float(ans2)
    ans = math.sqrt(dans2)

    # Output result
    print(ans)

if __name__ == "__main__":
    main()

